src/test/java/com/example/demo/config/TestConfig.java

```diff
@@ -2,18 +2,10 @@
 
 import org.springframework.boot.test.context.TestConfiguration;
 import org.springframework.context.annotation.Bean;
-import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
-import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
-
-import javax.sql.DataSource;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
 
 @TestConfiguration
 public class TestConfig {
-    
-    @Bean
-    public DataSource dataSource() {
-        return new EmbeddedDatabaseBuilder()
-            .setType(EmbeddedDatabaseType.H2)
-            .build();
-    }
+    // Add any test-specific beans here if needed
```

Comment:
could you add unit tests? You can submit a different PR.
3. Write Unit Tests for Controllers
	•	Use @WebMvcTest to load only the web layer components.
	•	Use MockMvc to test HTTP requests and responses without starting the server.

4. Write Unit Tests for Service Layer
	•	Test the service logic in isolation using mock objects for repositories or external services.
	•	Use Mockito for mocking dependencies.

5. Write Unit Tests for Repository Layer
	•	Use @DataJpaTest for repository tests, which loads an in-memory database like H2.